### PR TITLE
Update to filament 1.15.1

### PR DESCRIPTION
**Pull request by errissa**
All apparent issues with the upgrade have been resolved via code fixes and changes to Filament header files which are being maintained in the isl-org fork of Filament [here](https://github.com/isl-org/filament).

---
This change is [Reviewable](https://reviewable.io/reviews/isl-org/open3d/4475)

### CODE REVIEW

**Comment by yxlao** [comment]
Compiles fine on Apple Silicon. Still have the strange normal issue when running `python examples/python/gui/draw.py`

image


image

File: 3rdparty/filament/filament_build.cmake [hunk 1]
```diff
@@ -39,15 +41,21 @@ if(APPLE)
...
 set(lib_byproducts ${filament_LIBRARIES})
 list(TRANSFORM lib_byproducts PREPEND ${FILAMENT_ROOT}/${lib_dir}/${CMAKE_STATIC_LIBRARY_PREFIX})
 list(TRANSFORM lib_byproducts APPEND ${CMAKE_STATIC_LIBRARY_SUFFIX})
 
 ExternalProject_Add(
     ext_filament
     PREFIX filament
-    URL https://github.com/isl-org/filament/archive/d1d873d27f43ba0cee1674a555cc0f18daac3008.tar.gz
-    URL_HASH SHA256=00c3f41af0fcfb2df904e1f77934f2678d943ddac5eb889788a5e22590e497bd
+    GIT_REPOSITORY https://github.com/isl-org/filament
```
Comment (yxlao) [inline 3rdparty/filament/filament_build.cmake:55]: - Pin the commit id
- Download `tar.gz` instead of using git clone, so that the download can be cached.

**Review by yxlao — CHANGES_REQUESTED** [review]
I ran into https://github.com/isl-org/Open3D/issues/1909 on Ubuntu 20.04 with GCC 9.3.0, when:
```
cmake -DBUILD_FILAMENT_FROM_SOURCE=OFF ..
```
The error was:
```

```
when linking ``.

However,
```
cmake -DBUILD_FILAMENT_FROM_SOURCE=ON ..
```
works. 

- Was filament compiled on 18.04 (we shall do that on 18.04)?
- Did we use the `-fno-builtin` flag as suggested by `1909`?

File: 3rdparty/filament/filament_build.cmake (hunk 1 above)
Comment (errissa) [reply 3rdparty/filament/filament_build.cmake]: Done.

**Comment by errissa** [comment]
@yxlao The pre-built binary package has been rebuilt with `-fno-builtin` and should now work for both Ubuntu 18.04 and 20.04

**Comment by yxlao** [comment]
With Ubuntu 20.04 with GCC 9.3.0,
```
cmake -DBUILD_FILAMENT_FROM_SOURCE=OFF ..
```

Some new errors:
```
CubemapUtils.cpp:(.text._ZN8filament3ibl12CubemapUtils10solidAngleEmmm+0xf5): undefined reference to `__sqrtf_finite'
/usr/bin/ld: CubemapUtils.cpp:(.text._ZN8filament3ibl12CubemapUtils10solidAngleEmmm+0x102): undefined reference to `__atan2f_finite'
/usr/bin/ld: CubemapUtils.cpp:(.text._ZN8filament3ibl12CubemapUtils10solidAngleEmmm+0x133): undefined reference to `__sqrtf_finite'
/usr/bin/ld: CubemapUtils.cpp:(.text._ZN8filament3ibl12CubemapUtils10solidAngleEmmm+0x13f): undefined reference to `__atan2f_finite'
/usr/bin/ld: CubemapUtils.cpp:(.text._ZN8filament3ibl12CubemapUtils10solidAngleEmmm+0x189): undefined reference to `__sqrtf_finite'
/usr/bin/ld: CubemapUtils.cpp:(.text._ZN8filament3ibl12CubemapUtils10solidAngleEmmm+0x196): undefined reference to `__atan2f_finite'
/usr/bin/ld: CubemapUtils.cpp:(.text._ZN8filament3ibl12CubemapUtils10solidAngleEmmm+0x1c4): undefined reference to `__sqrtf_finite'
/usr/bin/ld: CubemapUtils.cpp:(.text._ZN8filament3ibl12CubemapUtils10solidAngleEmmm+0x1d0): undefined reference to `__atan2f_finite'
```

See the full logs:
https://asciinema.org/a/bPheL4KlaT1q8D1Ybdgaa91wS

**Comment by errissa** [comment]
Out of date.

**Pull request closed without merging**